splunklib/searchcommands/search_command.py

```diff
@@ -683,6 +686,21 @@ def _process_protocol_v2(self, argv, ifile, ofile):
         # noinspection PyBroadException
         try:
             self._record_writer = RecordWriterV2(ofile, getattr(self._metadata.searchinfo, 'maxresultrows', None))
+
+            # fix dvpl-7207: send reply to splunk and wait for reply of execute action
```

Comment:
There's something troubling I find about this comment, I'm just not sure that I can see any logic issues in the code yet.

Basically, the concern is this: it assumes that the SDK sends a chunk to splunkd and splunkd REPLIES to that chunk with something else.

However, splunkd should be the one initiating every action. splunkd sends the SDK a chunk (which is, effectively, a command to go do something), and the SDK replies with a response. In all cases, the SDK should be waiting for that command (e.g. either "getinfo" or "execute") from splunkd before doing anything. That is, there must be a 1:1 relationship between every chunk splunkd sends and every chunk the SDK replies with.

If there's ever an inversion of that notion (e.g. if the SDK is sending a chunk and waiting for a response from splunkd), it is very likely to lead to a protocol violation from the two getting out of lock-step.